### ISSUE TITLE
Allow linking to search results by using a URL parameter for search terms

### ISSUE THREAD

**Issue by quinncomendant** [mastodon/mastodon#21073]
### Pitch

Currently, if you want to search, you must open the search page at `https://example.com/search` and then enter your search query in the search box. I propose using a URL parameter for search results so opening a URL will load the search results directly, e.g., `https://example.com/search?q=[search-terms]`.  

### Motivation

This will allow sharing links to search results and also allow integrating search functionality into apps such as LaunchBar, Alfred, Quicksilver, and Raycast.

**Comment by colelawrence** [comment]
I'm interested in something like this as well to help resolve the friction of finding/following people on separate instances.

Otherwise, there is an incentive to join larger growing servers because there's less friction to follow their users.

In addition to being useful for launch bars and commander apps, I would like this functionality, to create a simple Mastodon link router and web extension to help people follow anyone in two clicks, rather than 2 or more (which could end up including someone having to type out their host server esp for newcomers who just started learning how the system works).

**Comment by rixx** [comment]
Yes, absolutely. I want to be able to have a browser shortcut to look up a user or a link immediately. Opening the page, waiting for full load, typing `s`, then pasting the query term is very clunky compared to being able to type `mastodon <search>` in my browser.

**Comment by colelawrence** [comment]
Originally, I just wanted to make it easier to create universal links, which I found a few ways around it without requiring searching. I put together [tooot.to](https://tooot.to/) which basically allows you to put `tooot.to/` before any profile URL or Mastodon handle to get a fast redirection page to the item.

Mastodon could improve on this by using some similar patterns I used like being able to type `<my-instance.tld>/` before any other Mastodon URL found in the wild which could automatically redirect to that item within the server. Here's [the source code for how I did this kind of redirection](https://github.com/colelawrence/tooot.to/blob/main/routes/_404.tsx).

**Comment by jasonsnell** [comment]
I stumbled upon this issue while trying to build a redirect pattern that would allow me to automatically open links to posts on other instances in my own instance's web UI.

`https://[other-instance]/@[user]/[post-id]`

...is easily rewritten, but since post IDs apparently chance across instances, you can't simply rewrite that URL to reference your own instance.

It would sure be nice if the Mastodon web interface would accept, from an already-logged-in user, something like:

`/search/@[user]@[other-instance]/[post-id]`

...properly recognize that as a search link to an external post, and send the user to the search page.

**Comment by salty-horse** [comment]
If and when this is implemented, please also add [OpenSearch](https://developer.mozilla.org/en-US/docs/Web/OpenSearch) metadata for better browser integration.

**Comment by pymander** [comment]
> For example, https://tooot.to/@fasterthanlime@octodon.social/109387627695106359
> Redirects me to my server at hachyderm like https://hachyderm.io/@fasterthanlime@octodon.social/109387627695106359

Unfortunately, the post ID changes between Mastodon servers, so this solution doesn't work.
